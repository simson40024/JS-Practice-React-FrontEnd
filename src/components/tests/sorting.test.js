import Sorting from "../shopping/sorting";
import { createRoot } from "react-dom/client";
import { act } from "@testing-library/react";
import * as hooks from "../../hook";

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  container.remove();
  container = null;
});

const mockDispatch = jest.spyOn(hooks, "useAppDispatch");

describe("sorting", () => {
  it("shoud create Sorting", () => {
    mockDispatch.mockReturnValue(jest.fn());
    act(() => {
      createRoot(container).render(<Sorting />);
    });
    expect(container).toMatchSnapshot();
  });
});
