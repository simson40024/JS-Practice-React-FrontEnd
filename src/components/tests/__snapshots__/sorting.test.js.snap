// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sorting shoud create Sorting 1`] = `
<div>
  <div
    class="sorting"
  >
    <div
      class="sorting-name"
    >
      by product:
      <select
        id="sort-by-product"
        name="sort-by-product"
      >
        <option
          selected=""
          value="-"
        >
          -
        </option>
        <option
          value="increase"
        >
          increase
        </option>
        <option
          value="decrease"
        >
          decrease
        </option>
      </select>
    </div>
    <div
      class="sorting-name"
    >
      by price:
      <select
        id="sort-by-price"
        name="sort-by-price"
      >
        <option
          selected=""
          value="-"
        >
          -
        </option>
        <option
          value="increase"
        >
          increase
        </option>
        <option
          value="decrease"
        >
          decrease
        </option>
      </select>
    </div>
    <div
      class="product-on-page"
    >
      amount on page:
      <select
        id=""
        name=""
      >
        <option
          selected=""
          value="10"
        >
          10
        </option>
        <option
          value="20"
        >
          20
        </option>
        <option
          value="30"
        >
          30
        </option>
      </select>
    </div>
  </div>
</div>
`;
